stdlib/public/core/Optional.swift

```diff
@@ -304,6 +304,39 @@ func _diagnoseUnexpectedNilOptional(_filenameStart: Builtin.RawPointer,
     line: UInt(_line))
 }
 
+/// Returns a Boolean value indicating whether two optional instances are
+/// equal.
+///
+/// Use this equal-to operator (`==`) to compare any two optional instances of
+/// a type that conforms to the `Equatable` protocol. The comparison returns
+/// `true` if both arguments are `nil` or if the two arguments wrap values
+/// that are equal. Conversely, the comparison returns `false` if only one of
+/// the arguments is `nil` or if the two arguments wrap values that are not
+/// equal.
+///
+///     let group1 = [1, 2, 3, 4, 5]
+///     let group2 = [1, 3, 5, 7, 9]
+///     if group1.first == group2.first {
+///         print("The two groups start the same.")
+///     }
+///     // Prints "The two groups start the same."
+///
+/// You can also use this operator to compare a non-optional value to an
+/// optional that wraps the same type. The non-optional value is wrapped as an
+/// optional before the comparison is made. In this example, the
+/// `numberToMatch` constant is wrapped as an optional before comparing to the
+/// optional `numberFromString`:
+///
+///     let numberToFind: Int = 23
+///     let numberFromString: Int? = Int("23")      // Optional(23)
+///     if numberToFind == numberFromString {
```

Comment:
I wonder if showing comparison to a literal would also help people "get" this i.e. `numberFromString == 23`